google/oauth2/service_account.py

```diff
@@ -81,13 +83,20 @@
 from google.auth import metrics
 from google.oauth2 import _client
 
+_LOGGER = logging.getLogger(__name__)
+
 _DEFAULT_TOKEN_LIFETIME_SECS = 3600  # 1 hour in seconds
 _GOOGLE_OAUTH2_TOKEN_ENDPOINT = "https://oauth2.googleapis.com/token"
+_TRUST_BOUNDARY_LOOKUP_ENDPOINT = (
+    "iamcredentials.{}/v1/projects/-/serviceAccounts/{}/trustBoundary"
+)
+TRUST_BOUNDARY_LOOKUP_ENABLED_ENV = "TRUST_BOUNDARY_LOOKUP_ENABLED"
```

Comment:
can you add a comment explaining what this is?

Reply:
Is this documented in an AIP?

Reply:
LOOUP_ENABLED check is a temporary solution and will be totally removed shortly.

Reply:
Can you expand on this a bit more? Once we release this, we can't remove it

Reply:
Yeah, the backend team want to launch trust boundary injection first at this stage. And the lookup must be bypassed. But in the end they want the attempt lookup in the refresh flow. The `TRUST_BOUNDARY_LOOKUP_ENABLED` is just a temporary solution and would be removed once the lookup endpoint is ready.

Reply:
Why not just not do the lookup if trust boundary info is provided when the credentials are created? What is the environment variable for?

Reply:
When the credential instance created we don't yet have a valid access_token. Without a valid access token we cannot do basic auth for lookup trust boundary.